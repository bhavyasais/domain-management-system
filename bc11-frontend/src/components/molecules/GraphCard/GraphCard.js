import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Paper, Typography } from '@material-ui/core';
import { Area, AreaChart, ResponsiveContainer } from 'recharts';
import theme from '../../../theme';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: '291px',
    height: '129px',
    borderRadius: '4px',
    flexGrow: 1,
    border: 'solid 1px #eeeeee',
  },
  graphType: {
    width: '104px',
    height: '24px',
    color: '#5c5c5c',
    fontSize: '16px',
    padding: '30px 156.8px 75px 30.2px',
  },
  totalCount: {
    width: '34px',
    height: '35px',
    color: theme?.palette?.black?.main,
    fontSize: '30px',
    margin: '-64px 30.2px',
    fontWeight: 'normal',
    backgroundColor: theme?.palette?.white?.main,
  },
  additionalCount: {
    width: '16px',
    height: '16px',
    fontSize: '14px',
    padding: '44px 197.7px 39px 77.3px',
    color: theme?.palette?.secondary?.main,
  },
  image: {
    transform: 'translate(155px, -80px)',
  },
}));
const GraphCard = ({ totalCount, additionalCount, domainValue, graphType }) => {
  const classes = useStyles();
  return (
    <Paper className={classes.root} variant="outlined">
      <Grid container direction="row" alignItems="center" justify="space-evenly">
        <Grid item>
          <Grid container direction="column">
            <Grid item>
              <Typography variant="h5" className={classes.graphType}>
                {graphType}
              </Typography>
            </Grid>
            <Grid container direction="row">
              <Grid item>
                <Typography variant="h1" className={classes.totalCount}>
                  {totalCount}
                </Typography>
                <Grid item className={classes.additionalCount}>
                  <Typography variant="caption">+{additionalCount}</Typography>
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
        {domainValue?.data && (
          <Grid item xs={12} className={classes.image}>
            <ResponsiveContainer width={105.6} height={36.2}>
              <AreaChart data={domainValue?.data}>
                <Area
                  dataKey="value"
                  fill={
                    graphType === 'Do Not Trust'
                      ? theme?.palette?.error?.main
                      : graphType === 'Trusted'
                      ? theme?.palette?.secondary?.main
                      : theme?.palette?.primary?.main
                  }
                  fillOpacity={0.1}
                  stroke={
                    graphType === 'Do Not Trust'
                      ? theme?.palette?.error?.main
                      : graphType === 'Trusted'
                      ? theme?.palette?.secondary?.main
                      : theme?.palette?.primary?.main
                  }
                  strokeWidth={2}
                />
              </AreaChart>
            </ResponsiveContainer>
          </Grid>
        )}
      </Grid>
    </Paper>
  );
};
export default GraphCard;
