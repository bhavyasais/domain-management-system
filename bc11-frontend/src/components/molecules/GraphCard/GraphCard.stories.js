import React, { useState } from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import GraphCard from './GraphCard';

export default {
  component: GraphCard,
  title: 'Molecules/GraphCard',
  decorators: [withKnobs],
};

export const graphCard = () => {
  const [item, setItem] = useState({
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: 'Mon', value: 30 },
      { date: 'Tue', value: 15 },
      { date: 'Wen', value: 12 },
      { date: 'Thu', value: 40 },
      { date: 'Fri', value: 20 },
      { date: 'Sat', value: 5 },
      { date: 'Sun', value: 30 },
    ],
  });
  return (
    <GraphCard
      graphType={text('Card Header', 'Trusted')}
      totalCount={text('Total Count', '30')}
      additionalCount={text('Additional Count', '1')}
      domainValue={item}
    />
  );
};
