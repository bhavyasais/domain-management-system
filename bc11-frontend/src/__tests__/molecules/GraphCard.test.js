import React from 'react';
import renderer from 'react-test-renderer';
import GraphCard from '../../components/molecules/GraphCard/GraphCard';

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('GraphCard', () => {
  it('Renders correctly', () => {
    let wrapper = renderer.create(
      <GraphCard
        graphType="Trusted"
        totalCount="30"
        additionalCount="1"
        domainValue={{
          value: 'zemoso',
          name: 'zemoso',
          data: [
            { date: 'Mon', value: 30 },
            { date: 'Tue', value: 15 },
            { date: 'Wen', value: 12 },
            { date: 'Thu', value: 40 },
            { date: 'Fri', value: 20 },
            { date: 'Sat', value: 5 },
            { date: 'Sun', value: 30 },
          ],
        }}
      />,
      { createNodeMock },
    );
    expect(wrapper).toMatchSnapshot();
  });
});
