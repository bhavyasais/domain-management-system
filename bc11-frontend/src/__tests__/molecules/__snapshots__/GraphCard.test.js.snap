// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphCard Renders correctly 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h5
            className="MuiTypography-root makeStyles-graphType-2 MuiTypography-h5"
          >
            Trusted
          </h5>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h1
              className="MuiTypography-root makeStyles-totalCount-3 MuiTypography-h1"
            >
              30
            </h1>
            <div
              className="MuiGrid-root makeStyles-additionalCount-4 MuiGrid-item"
            >
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                +
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-image-5 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="recharts-responsive-container"
        style={
          Object {
            "height": 36.2,
            "maxHeight": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "width": 105.6,
          }
        }
      >
        <div
          className="recharts-wrapper"
          style={
            Object {
              "cursor": "default",
              "height": 36.2,
              "position": "relative",
              "width": 105.6,
            }
          }
        >
          <svg
            className="recharts-surface"
            height={36.2}
            version="1.1"
            viewBox="0 0 105.6 36.2"
            width={105.6}
          >
            <defs>
              <clipPath
                id="recharts2-clip"
              >
                <rect
                  height={26.200000000000003}
                  width={95.6}
                  x={5}
                  y={5}
                />
              </clipPath>
            </defs>
            <g
              className="recharts-layer recharts-area"
            >
              <g
                className="recharts-layer"
              >
                <defs>
                  <clipPath
                    id="animationClipPath-recharts-area-3"
                  >
                    <rect
                      height={33}
                      width={0}
                      x={5}
                      y={0}
                    />
                  </clipPath>
                </defs>
                <g
                  className="recharts-layer"
                  clipPath="url(#animationClipPath-recharts-area-3)"
                >
                  <g
                    className="recharts-layer"
                    clipPath={null}
                  >
                    <path
                      className="recharts-curve recharts-area-area"
                      d="M5,11.55L20.93333333333333,21.375L36.86666666666666,23.34L52.8,5L68.73333333333332,18.1L84.66666666666666,27.925000000000004L100.6,11.55L100.6,31.200000000000003L84.66666666666666,31.200000000000003L68.73333333333332,31.200000000000003L52.8,31.200000000000003L36.86666666666666,31.200000000000003L20.93333333333333,31.200000000000003L5,31.200000000000003Z"
                      fill="#6dd230"
                      fillOpacity={0.1}
                      height={26.200000000000003}
                      stroke="none"
                      strokeWidth={2}
                      width={95.6}
                    />
                    <path
                      className="recharts-curve recharts-area-curve"
                      d="M5,11.55L20.93333333333333,21.375L36.86666666666666,23.34L52.8,5L68.73333333333332,18.1L84.66666666666666,27.925000000000004L100.6,11.55"
                      fill="none"
                      fillOpacity={0.1}
                      height={26.200000000000003}
                      stroke="#6dd230"
                      strokeWidth={2}
                      width={95.6}
                    />
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
        <div
          style={
            Object {
              "display": "none",
              "height": 0,
              "position": "absolute",
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
